message DCCLFieldOptions
{
  // what encoder / decoder should we use?
  optional string codec = 1 [default = "_default"];
  // leave this field out of DCCL tranmissions (has_field() will be false on receipt)
  optional bool omit = 2 [default = false];
  // set true for fields in the header (will *not* be encrypted)
  optional bool in_head = 3 [default = false];
  
  
  // double, float
  optional int32 precision = 4 [default = 0];
  // int, double, float
  optional double min = 5;
  optional double max = 6;
  optional double max_delta = 7;

  // static
  optional string static_value = 8;
  
  // string, bytes
  optional uint32 max_length = 9;  

  // any `repeated` field
  optional uint32 max_repeat = 10;
}

message DCCLMessageOptions
{
  required int32 id = 1;
  required uint32 max_bytes = 2;

  optional string codec = 3 [default = "_default"];
}

message QueueFieldOptions
{
  optional bool is_src = 1 [default = false];
  optional bool is_dest = 2 [default = false];
  optional bool is_time = 3 [default = false];
}

message QueueMessageOptions
{
  optional bool ack = 1 [default = true];  
  optional uint32 blackout_time = 2 [default = 0];
  optional uint32 max_queue = 3 [default = 0];
  optional bool newest_first = 4 [default = true];
  optional int32 ttl = 5 [default = 1800];
  optional double value_base = 6 [default = 1];
  optional bool encode_on_demand = 7 [default = false];
  optional double on_demand_skew_seconds = 8 [default = 1];
}
